### PR TITLE
Correct the comment for the priority of options and environments on otlptracegrpc

### PR DESCRIPTION
**Pull request by XSAM**
part of #5379

### CODE REVIEW

**Review by dmathieu — APPROVED** [review]

File: exporters/otlp/otlptrace/otlptracegrpc/options.go [hunk 1]
```diff
@@ -59,8 +59,9 @@ func WithInsecure() Option {
 //
 // If the OTEL_EXPORTER_OTLP_ENDPOINT or OTEL_EXPORTER_OTLP_TRACES_ENDPOINT
 // environment variable is set, and this option is not passed, that variable
-// value will be used. If both are set, OTEL_EXPORTER_OTLP_TRACES_ENDPOINT
-// will take precedence.
+// value will be used. If both environment variable are set,
```
Comment (dashpole) [inline exporters/otlp/otlptrace/otlptracegrpc/options.go:62]:
```suggestion
// value will be used. If both environment variables are set,
```

File: exporters/otlp/otlptrace/otlptracegrpc/options.go [hunk 2]
```diff
@@ -79,8 +80,9 @@ func WithEndpoint(endpoint string) Option {
...
```
Comment (dashpole) [inline exporters/otlp/otlptrace/otlptracegrpc/options.go:83]:
```suggestion
// value will be used. If both environment variables are set,
```

**Review by dashpole — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -37,6 +37,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Document instrument name requirements in `go.opentelemetry.io/otel/metric`. (#5435)
 - Prevent random number generation data-race for experimental rand exemplars in `go.opentelemetry.io/otel/sdk/metric`. (#5456)
 - Fix counting number of dropped attributes of `Record` in `go.opentelemetry.io/otel/sdk/log`. (#5464)
+- Correct comments for the priority of options and environments in `WithEndpoint` and `WithEndpointURL` in `go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc`. (#5508)
```
Comment (MrAlias) [inline CHANGELOG.md:40]:
```suggestion
- Correct comments for the priority of the `WithEndpoint` and `WithEndpointURL` options and their coresponding environment variables in  in `go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc`. (#5508)
```

File: exporters/otlp/otlptrace/otlptracegrpc/options.go [hunk 4]
```diff
@@ -59,8 +59,9 @@ func WithInsecure() Option {
 //
 // If the OTEL_EXPORTER_OTLP_ENDPOINT or OTEL_EXPORTER_OTLP_TRACES_ENDPOINT
 // environment variable is set, and this option is not passed, that variable
-// value will be used. If both are set, OTEL_EXPORTER_OTLP_TRACES_ENDPOINT
-// will take precedence.
+// value will be used. If both environment variables are set,
+// OTEL_EXPORTER_OTLP_TRACES_ENDPOINT will take precedence. If the environment
```
Comment (MrAlias) [inline exporters/otlp/otlptrace/otlptracegrpc/options.go:63]:
```suggestion
// OTEL_EXPORTER_OTLP_TRACES_ENDPOINT will take precedence. If an environment
```

File: exporters/otlp/otlptrace/otlptracegrpc/options.go [hunk 5]
```diff
@@ -79,8 +80,9 @@ func WithEndpoint(endpoint string) Option {
...
```
Comment (MrAlias) [inline exporters/otlp/otlptrace/otlptracegrpc/options.go:84]:
```suggestion
// OTEL_EXPORTER_OTLP_TRACES_ENDPOINT will take precedence. If an environment
```

File: exporters/otlp/otlptrace/otlptracegrpc/client_unit_test.go [hunk 6]
```diff
@@ -202,3 +202,75 @@ func TestExportContextLinksStopSignal(t *testing.T) {
...
+			envs: map[string]string{
+				"OTEL_EXPORTER_OTLP_ENDPOINT":        "foo2",
+				"OTEL_EXPORTER_OTLP_TRACES_ENDPOINT": "bar2",
+			},
+			options: []Option{
+				WithEndpointURL("http://bar:8080/path"),
+				WithEndpoint("foo"),
+			},
+			want: "foo",
+		},
```
Comment (MrAlias) [inline exporters/otlp/otlptrace/otlptracegrpc/client_unit_test.go:262]: Missing test for `WithEndpointURL` env var precidence.

```suggestion
		{
			name: "both options and envs",
			envs: map[string]string{
				"OTEL_EXPORTER_OTLP_ENDPOINT":        "foo2",
				"OTEL_EXPORTER_OTLP_TRACES_ENDPOINT": "bar2",
			},
			options: []Option{
				WithEndpointURL("http://bar:8080/path"),
				WithEndpoint("foo"),
			},
			want: "foo",
		},
		{
			name: "both options and envs",
			envs: map[string]string{
				"OTEL_EXPORTER_OTLP_ENDPOINT":        "foo2",
				"OTEL_EXPORTER_OTLP_TRACES_ENDPOINT": "bar2",
			},
			options: []Option{
				WithEndpoint("foo"),
				WithEndpointURL("http://bar:8080/path"),
			},
			want: "bar:8080",
		},
```

**Review by hanyuancheung — APPROVED** [review]

**Review by hanyuancheung — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Pull request merged**